### PR TITLE
[linux] powermanagement: change logind detection to also use Dbus

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

Fall back to checking Dbus for logind existence.

## Motivation and Context

On some environments (e.g. on flatpak) "/run/systemd/seats/" doesn't exist, so logind detection fails and powermanagement is not working.
see https://github.com/flathub/tv.kodi.Kodi/pull/20

## How Has This Been Tested?

on ubuntu, with and without flatpak

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp [hunk 1]
```diff
@@ -121,7 +121,38 @@ bool CLogindUPowerSyscall::HasLogind()
...
+    {
+      DBusMessageIter strIter;
+      dbus_message_iter_recurse(&structIter, &strIter);
+      if (dbus_message_iter_get_arg_type(&strIter) == DBUS_TYPE_STRING)
+      {
+        char *seat;
+        dbus_message_iter_get_basic(&strIter, &seat);
+        if (StringUtils::StartsWith(seat, "seat"))
+        {
+            CLog::Log(LOGDEBUG, "LogindUPowerSyscall::HasLogind - found seat: %s", seat);
```
Comment (pkerling) [inline xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp]: better to use `{}`

File: xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp [hunk 2]
```diff
@@ -121,7 +121,38 @@ bool CLogindUPowerSyscall::HasLogind()
...
+    return true;
+
+  // on some environments "/run/systemd/seats/" doesn't exist, e.g. on flatpak. Try DBUS instead.
+  CDBusMessage message(LOGIND_DEST, LOGIND_PATH, LOGIND_IFACE, "ListSeats");
+  DBusMessage *reply = message.SendSystem();
+  if (!reply)
+    return false;
+
+  DBusMessageIter arrIter;
+  dbus_message_iter_init(reply, &arrIter);
```
Comment (pkerling) [inline xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp]: Check return value

**Review by yol — APPROVED** [review]
lgtm apart from the minor

File: xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp [hunk 3]
```diff
@@ -121,7 +121,38 @@ bool CLogindUPowerSyscall::HasLogind()
...
+    dbus_message_iter_recurse(&arrIter, &structIter);
+    if (dbus_message_iter_get_arg_type(&structIter) == DBUS_TYPE_STRUCT)
+    {
+      DBusMessageIter strIter;
+      dbus_message_iter_recurse(&structIter, &strIter);
+      if (dbus_message_iter_get_arg_type(&strIter) == DBUS_TYPE_STRING)
+      {
+        char *seat;
+        dbus_message_iter_get_basic(&strIter, &seat);
+        if (StringUtils::StartsWith(seat, "seat"))
```
Comment (pkerling) [inline xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp]: Why is this necessary?

Comment (wsnipex) [reply xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp]: it isn't absolutely necessary, we could probably get away with just testing if the dbus call returns != nullptr, but I wanted to be precise and check if we really get a seat back.
We could relax this once we have some wider testing.

Comment (pkerling) [reply xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp]: IMO since we're already calling `ListSeats`, it would be sufficient to call `dbus_message_iter_get_array_length` and be happy if that is non-zero

Comment (pkerling) [reply xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp]: Not a blocker as long as you promise to remove it later ;-)
Depending on the results of course.
